src/coffee/product-import.coffee

```diff
@@ -8,4 +10,88 @@ class ProductImport
     @sync = new ProductSync
     @client = new SphereClient options
 
+  _resetSummary: ->
+    @_summary =
+      emptySKU: 0
+      created: 0
+      updated: 0
+
+
+  performStream: (chunk, cb) ->
+    @_processBatches(chunk).then -> cb()
+
+  _processBatches: (products) ->
+    batchedList = _.batchList(products, 30) # max parallel elem to process
+    Promise.map batchedList, (productsToProcess) =>
+      # extract all skus from master variant and variants of all jsons in the batch
+      skus = @_extractUniqueSkus(productsToProcess)
+      predicate = @_prepareProductFetchBySkuQueryPredicate(skus)
+      # Fetch products from product projections end point by list of skus.
+      @client.productProjections
+      .where(predicate)
+      .fetch()
+      .then (results) =>
+        debug 'Fetched products: %j', results
+        queriedEntries = results.body.results
+        @_createOrUpdate productsToProcess, queriedEntries
+      .then (results) =>
+        _.each results, (r) =>
+          switch r.statusCode
+            when 201 then @_summary.created++
+            when 200 then @_summary.updated++
+        Promise.resolve()
+    ,{concurrency: 1} # run 1 batch at a time
+
+
+  _prepareProductFetchBySkuQueryPredicate: (skus) ->
+    predicate = {}
+    skuString = "sku in (\"#{skus.join('", "')}\")"
+    predicate.predicateString = "masterVariant(#{skuString}) or variants(#{skuString})"
+    predicate.byteSize = Buffer.byteLength(predicate.predicateString,'utf-8')
```

Comment:
because the predicate might become too large if the number of variants is huge and the nginx server has a query limit of 8kb.
So we need to handle it if the query becomes larger than 8kb. Still have to figure out how.